AEPLifecycle/Sources/Lifecycle.swift

```diff
@@ -65,29 +67,42 @@ public class Lifecycle: NSObject, Extension {
 
         if event.isLifecycleStartEvent {
             Log.debug(label: LifecycleConstants.LOG_TAG, "Starting lifecycle.")
-            start(event: event, configurationSharedState: configurationSharedState)
+            startApplicationLifecycle(event: event, configurationSharedState: configurationSharedState)
         } else if event.isLifecyclePauseEvent {
             Log.debug(label: LifecycleConstants.LOG_TAG, "Pausing lifecycle.")
-            lifecycleState.pause(pauseDate: event.timestamp)
+            pauseApplicationLifecycle(event: event)
         }
     }
 
     // MARK: Helpers
 
+    /// Start the lifecycle session for standard and XDM workflows
     /// Invokes the start business logic and dispatches any shared state and lifecycle response events required
     /// - Parameters:
     ///   - event: the lifecycle start event
     ///   - configurationSharedState: the current configuration shared state
-    private func start(event: Event, configurationSharedState: SharedStateResult) {
+    private func startApplicationLifecycle(event: Event, configurationSharedState: SharedStateResult) {
+        let install = isInstall()
         let prevSessionInfo = lifecycleState.start(date: event.timestamp,
                                                    additionalContextData: event.additionalData,
                                                    adId: getAdvertisingIdentifier(event: event),
-                                                   sessionTimeout: getSessionTimeoutLength(configurationSharedState: configurationSharedState.value))
+                                                   sessionTimeout: getSessionTimeoutLength(configurationSharedState: configurationSharedState.value),
+                                                   isInstall: install)
         updateSharedState(event: event, data: lifecycleState.getContextData()?.toEventData() ?? [:])
 
         if let prevSessionInfo = prevSessionInfo {
             dispatchSessionStart(date: event.timestamp, contextData: lifecycleState.getContextData(), previousStartDate: prevSessionInfo.startDate, previousPauseDate: prevSessionInfo.pauseDate)
         }
+
+        lifecycleV2.start(date: event.timestamp, additionalData: event.additionalData, isInstall: install)
+    }
```

Comment:
Can we also move the logic to persist the install date here?

Reply:
good catch 👍